### PR TITLE
adding cards for WJetsToLNu samples 

### PR DESCRIPTION
**Pull request by deepakcern**
I would like to add cards for the WJetsToLNu sample with 3Jets.

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W3JetsToLNu_13TeV-madgraphMLM-pythia8/W3JetsToLNu_13TeV-madgraphMLM-pythia8_proc_card.dat [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import model sm-ckm_no_b_mass
+
+generate p p > w-, w- > l- vl~ /h @0
+add process p p > w- j, w- > l- vl~ @1
+add process p p > w- j j, w- > l- vl~ @2
+add process p p > w- j j j, w- > l- vl~ @3
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W3JetsToLNu_13TeV-madgraphMLM-pythia8/W3JetsToLNu_13TeV-madgraphMLM-pythia8_proc_card.dat]: as commented here
https://github.com/cms-sw/genproductions/pull/1474
are you sure you want to include 0j,1j,2j ?
why do you include only the W minus?!?

Comment (deepakcern) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W3JetsToLNu_13TeV-madgraphMLM-pythia8/W3JetsToLNu_13TeV-madgraphMLM-pythia8_proc_card.dat]: Sorry for mistake, i talked to Anirban and then compare with previous cards https://github.com/cms-sw/genproductions/tree/pre2017/bin/MadGraph5_aMCatNLO/cards/production/13TeV/WJets_jetbinning_LO_MLM 
and i found that we need to include W+ and W- , we don't need to include 0j, 1j and 2j.
I will make some changes according to previous cards.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat [hunk 2]
```diff
@@ -130,9 +130,9 @@ $DEFAULT_PDF_MEMBERS = reweight_PDF     ! if pdlabel=lhapdf, this is the lhapdf
 # Maximum and minimum absolute rapidity (for max, -1 means no cut)   *
 #*********************************************************************
    5  = etaj    ! max rap for the jets 
-  2.5 = etab    ! max rap for the b
+  -1 = etab    ! max rap for the b
   -1  = etaa    ! max rap for the photons 
-  2.5 = etal    ! max rap for the charged leptons 
+  -1 = etal    ! max rap for the charged leptons
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: why did you change these parameters? they don't look ok to me

Comment (deepakcern) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: I checked with previous cards thats why i changed it.

Comment (deepakcern) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: should i put 2.5=etal?

Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: can you poiny me to the previous cards you refer to?

Comment (deepakcern) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: here you can see https://github.com/cms-sw/genproductions/tree/pre2017/bin/MadGraph5_aMCatNLO/cards/production/13TeV/WJets_jetbinning_LO_MLM

Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: this looks like a bug to me....
@kdlong @qliphy can you please give a feedback?

Comment (anirban1982) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: Hi Luca,

Deepak is mentioning the following pre2017 tag card, from where he started changing,

https://github.com/cms-sw/genproductions/blob/pre2017/bin/MadGraph5_aMCatNLO/cards/production/13TeV/WJets_jetbinning_LO_MLM/W1Jets_madgraph_5f_LO/W1Jets_madgraph_5f_LO_run_card.dat

The previous cards were made by Deepak, and now I suggested him to start from the above link, which was used for 92X generation.

Comment (kdlong) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: I think we would normally want a sample like this to be as inclusive as possible (e.g. no eta cut on the leptons) so that they can be used as backgrounds for all analyses.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat [hunk 3]
```diff
@@ -48,9 +48,8 @@
 #*********************************************************************
 # PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
 #*********************************************************************
- 'lhapdf'    = pdlabel     ! PDF set                                  
-$DEFAULT_PDF_SETS = lhaid
-$DEFAULT_PDF_MEMBERS = reweight_PDF     ! if pdlabel=lhapdf, this is the lhapdf number 
+ lhapdf = pdlabel ! PDF set
+ 306000 = lhaid
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: I am confused now, why did you revert the lhaid?

Comment (anirban1982) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: Hi Luca,

he did it according to the twiki for LO sample:

https://twiki.cern.ch/twiki/bin/viewauth/CMS/QuickGuideMadGraph5aMCatNLO#PDF_Choice_for_2017_production

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat (hunk 2 above)
Comment (anirban1982) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: @kdlong thanks.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat (hunk 3 above)
Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: I am afraid you need to go to the second paragraph...
also cfr https://twiki.cern.ch/twiki/bin/view/CMS/Moriond18MC#Checks_to_the_datacards_BEFORE_T

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat (hunk 2 above)
Comment (deepakcern) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: Thanks

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat (hunk 3 above)
Comment (kdlong) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: Agreed that the twiki is confusing. Actually this syntax is supported at LO as well. Updating now.

Comment (anirban1982) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: T = pdfwgt this should also be included right?

Comment (efeyazgan) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: yes

Comment (anirban1982) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W2JetsToLNu_13TeV-madgraphMLM-pythia8/W2JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: Thanks @efeyazgan , and @kdlong for changing the twiki, it is clear now.  Review them please and submit the PR.

**Comment by deepakcern** [comment]
I have changed lhaid.
$DEFAULT_PDF_SETS = lhaid
$DEFAULT_PDF_MEMBERS = reweight_PDF

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat (hunk 2 above)
Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: @deepakcern it means that you need to revert the cuts to the previous values...

Comment (anirban1982) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat]: @perrozzi the inclusive cuts are already there, see, 

https://github.com/deepakcern/genproductions/blob/WJets/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/WJetsToLNu/W1JetsToLNu_13TeV-madgraphMLM-pythia8/W1JetsToLNu_13TeV-madgraphMLM-pythia8_run_card.dat#L135

@deepakcern has already reverted them back.
